### ISSUE TITLE
Courses: Console error when removing description.

### ISSUE THREAD

**Issue by jsschf** [open-learning-exchange/planet#6833]
You are allowed to submit even if the description is empty, but you get a console error and infinite loading.

https://www.loom.com/share/5cc52300b6b245ff9c62d943d966025c

![Annotation 2020-07-13 222307](https://user-images.githubusercontent.com/54686871/87380227-933e2d00-c581-11ea-8888-23c4d6ba27b2.png)

**Comment by paulbert** [comment]
The error is in `MarkdownService.filterMissingImages`.  I think if the markdown is blank it's probably supplying a `null` value instead of a string.

**Comment by paulbert** [comment]
There are also issues with having blank descriptions in steps.  I'll make a new issue **On Hold**.  If this one needs more than the above, it could be the same fix for both.

**Comment by paulbert** [comment]
In testing this morning I noticed we needed a special validator for markdown.  I'm not sure this entirely fixes the issue, and will test more once that is merged.

**Comment by paulbert** [comment]
The error is still showing up when going back and forth on steps as described in #6834

**Comment by singharpita** [comment]
@paulbert found that when switching between steps it makes nested markdown
`description: { images: [], text: { images: [], text: } }`
But could not figure out how to fix it as same function is used by course as well for converting description

**Comment by paulbert** [comment]
When we change a step the `images` should be concatenated with the `this.images` array and only the `text` should be saved as the `description`.
